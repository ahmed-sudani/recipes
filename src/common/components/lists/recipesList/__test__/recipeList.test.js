import { render, screen, within } from '@testing-library/react'
import { RecipesList } from '..'

const recipes = [
  {
    id: '1',
    name: 'chechen',
    time: '50:00',
    image: '/cster.png',
    country: 'italy',
    ingredients: [],
  },

  {
    id: '2',
    name: 'chechen',
    time: '50:00',
    image: '/cster.png',
    country: 'italy',
    ingredients: [],
  },
]

describe('Recipes list', () => {
  it('should render RecipeList component', () => {
    render(<RecipesList recipes={[]} />)
    const recipeListElement = screen.getByTestId('container')
    expect(recipeListElement).toMatchSnapshot()
  })
  it('should have 2 recipes in the list', () => {
    render(<RecipesList recipes={recipes} />)
    const recipeListElement = screen.getByTestId('container')
    const recipesList = within(recipeListElement).getAllByTestId('recipe')
    expect(recipesList.length).toEqual(2)
  })
})
