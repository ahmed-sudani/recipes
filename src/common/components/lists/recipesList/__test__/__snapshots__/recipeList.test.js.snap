// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes list should render RecipeList component 1`] = `
<div
  class="container"
  data-testid="container"
/>
`;
